packages/hierarchy-builder/src/test/internal/operators/Grouping.test.ts

```diff
@@ -0,0 +1,207 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { from } from "rxjs";
+import sinon from "sinon";
+import { LogLevel } from "@itwin/core-bentley";
+import { HierarchyNode } from "../../../hierarchy-builder/HierarchyNode";
+import {
+  createGroupingHandlers,
+  createGroupingOperator,
+  GroupingHandler,
+  GroupingHandlerResult,
+  LOGGING_NAMESPACE,
+} from "../../../hierarchy-builder/internal/operators/Grouping";
+import * as baseClassGrouping from "../../../hierarchy-builder/internal/operators/grouping/BaseClassGrouping";
+import * as classGrouping from "../../../hierarchy-builder/internal/operators/grouping/ClassGrouping";
+import * as groupHiding from "../../../hierarchy-builder/internal/operators/grouping/GroupHiding";
+import * as labelGrouping from "../../../hierarchy-builder/internal/operators/grouping/LabelGrouping";
+import { IMetadataProvider } from "../../../hierarchy-builder/Metadata";
+import { createTestNode, getObservableResult, setupLogging } from "../../Utils";
+
+describe("Grouping", () => {
+  const metadata = {} as unknown as IMetadataProvider;
+
+  before(() => {
+    setupLogging([{ namespace: LOGGING_NAMESPACE, level: LogLevel.Trace }]);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("createGroupingOperator", () => {
+    let applyGroupingHidingParamsStub: sinon.SinonStub;
+    beforeEach(() => {
+      applyGroupingHidingParamsStub = sinon.stub(groupHiding, "applyGroupHidingParams").callsFake((props) => props);
+    });
+
+    it("doesn't change input nodes when grouping handlers don't group", async () => {
+      const nodes = [
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.A", id: "0x1" }] },
+          label: "1",
+          params: {
+            grouping: {
+              byBaseClasses: {
+                fullClassNames: ["TestSchema.TestParentClassA", "TestSchema.TestParentClassAA"],
+              },
+              byClass: true,
+              byLabel: true,
+            },
+          },
```

Comment:
I don't think we need to pass grouping params - the grouping operator is using stubbed handlers which don't care about them. Applies to more tests below.

Reply:
Removed